### PR TITLE
Fix Firefox mouse wheel scrolling

### PR DESCRIPTION
**Pull request by my8bird**

### CODE REVIEW

File: src/vrs.Util.js [hunk 1]
```diff
@@ -27,36 +27,74 @@ Ext.format = Ext.util.Format.format;
...
+		   adjustScroller(scroller, evt);
+	   }
+   }
+
+   /**
+    * Adds an event mouse whell event listener so that the scroller can behave
+    * correctly on desktops
+    */
+   vrs.addMouseWheelHooks = function() {
+      document.addEventListener('wheel', onWheelEvent, false);
```
Comment (my8bird) [inline src/vrs.Util.js]: Changing the event is the real fix.  The rest is taking a code sample which was found online and cleaning it up

**Comment by my8bird** [comment]
P5 Checklist 
 * https://p5.tpondemand.com/entity/2488 
 
 - [ ] Release Note
 - [ ] Tests
 - [ ] Dependency versions sheet and license files updated

File: src/vrs.Util.js [hunk 2]
```diff
@@ -27,36 +27,74 @@ Ext.format = Ext.util.Format.format;
...
+          offset = {
+             x: 0,
+             y: evt.deltaY * multiplier
+          };
+
+		scroller.fireEvent('scrollstart', scroller, scroller.position.x, scroller.position.y, evt);
+		scroller.scrollBy(offset.x, offset.y);
+      console.log('asdf');
+		scroller.snapToBoundary();
+		scroller.fireEvent('scrollend', scroller, scroller.position.x, scroller.position.y - offset.y);
```
Comment (war1025) [inline src/vrs.Util.js]: nitpick: indentation / remove `console.log()`

File: src/vrs.Util.js [hunk 3]
```diff
@@ -27,36 +27,74 @@ Ext.format = Ext.util.Format.format;
...
+   }
+
+   function findScroller(el) {
+      var cmp, scroller;;
```
Comment (war1025) [inline src/vrs.Util.js]: nitpick: extra `;`

File: src/vrs.Util.js [hunk 4]
```diff
@@ -27,36 +27,74 @@ Ext.format = Ext.util.Format.format;
...
-	    }
-	 }
-	 _results.push(el = el.parentNode);
+         if (el && el.className && el.className.indexOf('x-container') >= 0) {
+            cmp = Ext.getCmp(el.id);
+	         if (cmp && typeof cmp.getScrollable == 'function' && cmp.getScrollable()) {
+	            scroller = cmp.getScrollable().getScroller();
+               if (scroller) {
+                  return scroller;
+               }
```
Comment (war1025) [inline src/vrs.Util.js]: nitpick: indentation above seems wonky.

**Comment by war1025** [comment]
+1
 - [ ] Seems like tabs and spaces are mixed. So decide whether that is something to worry about or not.

File: src/vrs.Util.js [hunk 5]
```diff
@@ -27,36 +27,74 @@ Ext.format = Ext.util.Format.format;
...
-      var el = e.target;
-      var offset, scroller, _results;
-      _results = [];
+ * Add hooks to intercept mouse wheel events.
+ *
+ * see for base concept:
+ * https://www.sencha.com/forum/showthread.php?110792-Add-support-for-mouse-scrolling-%28for-development%29&p=744700&viewfull=1#post744700
+ */
+(function() {
+   // `wheel` events can have different modes to indicate how what the delta means.
```
Comment (aronbierbaum) [inline src/vrs.Util.js]: nitpick: `indicate what the delta means.`

File: src/vrs.Util.js [hunk 6]
```diff
@@ -27,36 +27,74 @@ Ext.format = Ext.util.Format.format;
...
+    * Event handler for `wheel` events which will search up the DOM looking for
+    * a scroller and adjust the scrollers possition to simulate normal scrolling.
+    *
+    * @param evt https://developer.mozilla.org/en-US/docs/Web/Events/wheel
+    */
+   function onWheelEvent(evt) {
+      var scroller = findScroller(evt.target);
+      if (scroller) {
+		   adjustScroller(scroller, evt);
+	   }
```
Comment (aronbierbaum) [inline src/vrs.Util.js]: nitpick: Looks like there are more tabs in this file.

File: src/vrs.Util.js [hunk 7]
```diff
@@ -27,36 +27,74 @@ Ext.format = Ext.util.Format.format;
...
+   }
```
Comment (aronbierbaum) [inline src/vrs.Util.js]: The old code returned the elements that were discovered while looking for the container. Is it safe to assume that this was and isn't required?

**Comment by aronbierbaum** [comment]
+1
- [ ] Handle nitpicks
- [ ] Regression test if a test suite exists and a test can be added

File: src/vrs.Util.js (hunk 4 above)
Comment (my8bird) [inline src/vrs.Util.js]: Looks like this file always had tabs and spaces which makes sense it is a bunch of code copied out of various forum posts.

File: src/vrs.Util.js (hunk 7 above)
Comment (my8bird) [inline src/vrs.Util.js]: Yes, it was not required.  The code was posted to a forum asking if others could clean it up, :smile:

**Pull request merged**